docs/Reference/Errors.md

```diff
@@ -111,8 +111,11 @@ fastify.setErrorHandler(function (error, request, reply) {
     // Log error
     this.log.error(error)
     // Send error response
-    reply.status(500).send({ ok: false })
+    return reply.status(500).send({ ok: false })
   }
+  
+  // pop error to default error handler
+  return error
```

Comment:
Then, I would probably do in this way to reduce the `PromiseLike` around trip.

```suggestion
    reply.status(500).send({ ok: false })
  } else {
    reply.send(error)
  }
```

Reply:
This improve perf but it behavior differently, this doesn't pop error to parent error handler, for example an instance error handler, and a sub-instance error handler 🤔 as for document I would suggest just return or throw error.

Reply:
`return error` is actually doing the same as `reply.send(error)`. But the action is implicitly done by `fastify`.

See Line 69.
https://github.com/fastify/fastify/blob/90fb718ea8239e59b3abdf6906fb82554d8cc0db/lib/error-handler.js#L64-L71

Reply:
Oh, yes.

Maybe update document here too https://www.fastify.io/docs/latest/Reference/Server/#seterrorhandler ?